src/components/datepicker/js/datepickerDirective.js

```diff
@@ -402,9 +402,17 @@
 
     // Responds to external changes to the model value.
     self.ngModelCtrl.$formatters.push(function(value) {
+      var parsedValue = angular.isDefined(value) ? Date.parse(value) : null;
+
+      if (parsedValue) {
+        value = self.dateLocale.parseDate(parsedValue);
```

Comment:
Isn't this redundant with `Date.parse(value)` above? Also, `mdDateLocaleProvider`.`parseDate` should expect a string as well, and we're passing a `Date` object here.